graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/MethodDataFetcherFactory.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.internal
+
+import graphql.schema.DataFetcher
+import org.springframework.core.DefaultParameterNameDiscoverer
+import org.springframework.core.ParameterNameDiscoverer
+import java.lang.reflect.Method
+
+class MethodDataFetcherFactory(
```

Comment:
The reason I made this separate is because I noticed `DgsSchemaProvider` was accepting a bunch of parameters that were specifically for the `DataFetcherInvoker`, such as `InputObjectMapper`, `CookieValueResolver`, and `ParameterNameDiscoverer`. It felt cleaner to extract it into a class that knows how to create these DataFetchers given a method, in order to encapsulate and hide those implementation details. This way `DgsSchemaProvider` doesn't need to know the dependencies of `DataFetcherInvoker`, so if those happen to change it only needs to be updated in one place, this `MethodDataFetcherFactory` whose only responsibility is creating `DataFetcherInvoker` instances.

Reply:
Thanks for the explanation.

As for `@Bean` vs `@Import`, I have a slight preference for `@Bean` because it's a bit clearer (to me). Not a strong opinion though. @berngp @srinivasankavitha thoughts?

Reply:
I prefer @Bean simply because of familiarity. I haven't used @Import, but can get used to it :)

Reply:
In this case I prefer `@Bean`, the `@Import` annotation is meant to be used to group `@Configuraiton` classes so we have one that _includes_ the other ones. e.g. Let say you have a `MovieConfiguration`, a `TalentConfiguration`, etc. and you want to help your uses just load everything and don't have to remember each. You can then define a `ProductionConfiguration`, let's say, and `@Import` of all of the other related configurations there.

Reply:
🤷 I don't have a strong opinion, `@Import` does work for regular components as well, as is mentioned in the [Javadoc](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Import.html), but I can switch it to use a `@Bean` method for consistency.